File: VB/02_Text/AddBorderForText.vb
Imports Spire.Pdf
Imports Spire.Pdf.Graphics

Namespace AddBorderForText
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a pdf document
			Dim doc As New PdfDocument()

			' Load from file
			doc.LoadFromFile("..\..\..\..\..\..\Data\SampleB_1.pdf")

			' Get the first page
			Dim page As PdfPageBase = doc.Pages(0)

			' Define the text to be added to the page
			Dim text As String = "Hello, World!"

			' Define the font for the text
			Dim font As New PdfTrueTypeFont(New Font("Times New Roman", 11, FontStyle.Regular), True)


			' =============================================================================
			' Use the following code for netstandard dlls
			' =============================================================================
			' Dim font As New PdfTrueTypeFont("Times New Roman", 11, FontStyle.Regular, True)
			' =============================================================================

			' Measure the size of the text
			Dim size As SizeF = font.MeasureString(text)

			' Define the brush for drawing the text
			Dim brush As New PdfSolidBrush(Color.Black)

			' Define the position where the text will be drawn
			Dim x As Integer = 60
			Dim y As Integer = 300

			' Draw the text on the page
			page.Canvas.DrawString(text, font, New PdfSolidBrush(Color.Black), x, y)

			' Draw a rectangle border around the text
			page.Canvas.DrawRectangle(New PdfPen(brush, 0.5F), New Rectangle(x, y, CInt(size.Width), CInt(size.Height)))

			' Specify the file name for the resulting document
			Dim result As String = "AddBorderForText-result.pdf"

			' Save the document to a file
			doc.SaveToFile(result)

			' Launch the Pdf file
			PDFDocumentViewer(result)
		End Sub
		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try

		End Sub
	End Class
End Namespace

File: CS/02_Text/AddBorderForText.cs
using Spire.Pdf;
using Spire.Pdf.Graphics;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using static System.Net.Mime.MediaTypeNames;

namespace AddBorderForText
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Create a pdf instance
            PdfDocument doc = new PdfDocument();

            //Load from file
            doc.LoadFromFile(@"..\..\..\..\..\..\Data\SampleB_1.pdf");

             //Get the first page
            PdfPageBase page = doc.Pages[0];

            string text = "Hello, World!";

            //Create the font to use and set the font style 
            PdfTrueTypeFont font = new PdfTrueTypeFont(new Font("Times New Roman", 11, FontStyle.Regular),true);



            //////////////////Use the following code for netstandard dlls/////////////////////////
            /*
             PdfTrueTypeFont font = new PdfTrueTypeFont("Times New Roman", 11, PdfFontStyle.Regular, true);
            */

            //Measure the size of the text
            SizeF size = font.MeasureString(text);

            //Create a PdfSolidBrush instance for setting the color of text
            PdfSolidBrush brush = new PdfSolidBrush(Color.Black);
            int x = 60;
            int y = 300;

            //Draw the text on page
            page.Canvas.DrawString(text,
                                   font,
                                   new PdfSolidBrush(Color.Black),
                                   x, y);

            //Draw border for text          
            page.Canvas.DrawRectangle(new PdfPen(brush, 0.5f),new Rectangle(x, y, (int)size.Width, (int)size.Height));

            String result = "AddBorderForText-result.pdf";
            //save to file
            doc.SaveToFile(result);
            
            //Launch the Pdf file
            PDFDocumentViewer(result);

        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }                    
   
        }
    }
}
